pkg/cluster/kubernetes_cluster.go

```diff
@@ -190,12 +174,87 @@ func (c *KubernetesCluster) Connect(ctx context.Context, cmdOptions *ConnectArgu
 		return err
 	}
 
-	err = c.createKafkaConnectionCustomResource(ctx, currentNamespace, &kafkaInstance)
+	switch cmdOptions.SelectedService {
+	case "kafka":
+		err = c.checkAndCreateKafkaConnectionCustomResource(ctx, currentNamespace, cmdOptions.SelectedServiceID)
+		if err != nil {
+			return err
+		}
+	case "service-registry":
+		err = c.checkAndCreateServiceRegistryConnectionCustomResource(ctx, currentNamespace, cmdOptions.SelectedServiceID)
+		if err != nil {
+			return err
+		}
+	case "":
+		var selectedKafkaInstance string
+		var selectedRegistryInstance string
+
+		cfg, err := c.config.Load()
+		if err != nil {
+			return err
+		}
+
+		if cfg.Services.Kafka == nil || cmdOptions.IgnoreContext {
+			// nolint
+			selectedKafka, err := kafkaUtil.InteractiveSelect(c.connection, c.logger)
+			if err != nil {
+				return err
+			}
+			if selectedKafka == nil {
+				return nil
+			}
+			selectedKafkaInstance = selectedKafka.GetId()
+		} else {
+			selectedKafkaInstance = cfg.Services.Kafka.ClusterID
+		}
+
+		err = c.checkAndCreateKafkaConnectionCustomResource(ctx, currentNamespace, selectedKafkaInstance)
+		if err != nil {
+			return err
+		}
+
+		if cfg.Services.ServiceRegistry == nil || cmdOptions.IgnoreContext {
+			// nolint
+			selectedServiceRegistry, err := serviceregistry.InteractiveSelect(c.connection, c.logger)
+			if err != nil {
+				return err
+			}
+			if selectedServiceRegistry == nil {
+				return nil
+			}
+			selectedRegistryInstance = selectedServiceRegistry.GetId()
+		} else {
+			selectedRegistryInstance = cfg.Services.ServiceRegistry.InstanceID
+		}
+
+		err = c.checkAndCreateServiceRegistryConnectionCustomResource(ctx, currentNamespace, selectedRegistryInstance)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	return nil
+}
+
+func (c *KubernetesCluster) checkAndCreateKafkaConnectionCustomResource(ctx context.Context, namespace string, kafkaID string) error {
```

Comment:
It will be good if all kafka, registry logic were separated.
We could use strategy pattern to inject behaviour etc.